internal/driver/readhandler.go

```diff
@@ -103,7 +105,26 @@ func makeReadRequest(deviceClient *opcua.Client, req sdkModel.CommandRequest) (*
 
 	// make new result
 	reading := resp.Results[0].Value.Value()
-	return newResult(req, reading)
+	result, err := newResult(req, reading)
```

Comment:
Since we now don't override the origin value anynomre i would say it is ok to do this in the makeReadRequest method since we are adding additional info the the read value, not directly manipulating its core fields...like origin.

Reply:
If you capture the source timestamp or server timestamp just in the `makeReadRequest` method then this is missed in the `subscriptionlistener.go` mentioned above especially in its `onIncomingDataReceived` method. I guess that receiving the OPCUA datapoints is mostly and best done by "subscriptions" and that the `makeReadRequest` method could be ineffective for receiving subscribed values.

Reply:
moved the code to a seperate methode for easier use from different calling origins